spec/src/main/asciidoc/clientexamples.asciidoc

```diff
@@ -80,6 +80,49 @@ All built in HTTP methods are supported by the client API.  Likewise, all base p
 
 Users may specify the media (MIME) type of the outbound request using the `@Consumes` annotation (this determine the `Content-Type` HTTP header), and the expected media type(s) of the response by using the `@Produces` annotation (the `Accept` HTTP header).  This indicates that the client interface expects that the remote _service_ consumes/produces the specified types.
 
+=== Specifying Additional Client Headers
+
+While it is always possible to add a `@HeaderParam`-annotated method argument to specify headers, some times that does not make sense within the context of the application. For example, you may want to specify a username/password in the `Authorization` header to a secure remote service, but you may not want to have a `String authHeader` parameter in the client interface method.
+The `@ClientHeaderParam` annotation can allow users to specify HTTP headers that should be sent without altering the client interface method signature. The annotation contains three attributes: `name`, `value`. and `required`.  The name attribute is used to specify the header name. The value attribute is used to specify the value of the header. The value can be specified explicitly
+or it can reference a default method on the client interface that would compute the value of the header - in this latter case, the compute method name must be surrounded by curly braces. The compute method must be a default method on the interface, must return a `String` and must either contain no arguments or contain a single `String` argument - the implementation will use this
+`String` argument to pass the name of the header. The required attribute determines what should happen in the event that the compute method throws an exception. If the required attribute is set to true (default), then the client request will fail if the compute method throws an exception.  If it is set to false, then the client request will continue, but without sending the HTTP header.
```

Comment:
Do we need some kind of big flashing warning in the spec doc about the fact that required=true will cause an exception if the method call fails?

Reply:
Sure, I can bold that text.